### PR TITLE
Config IPv6 as persistent

### PR DESCRIPTION
**Pull request by juliayakovlev**
## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster_aws.py [hunk 1]
```diff
@@ -306,9 +302,51 @@ def network_config_ipv6_workaround_script():  # pylint: disable=invalid-name
...
+              version: 1
+              config:
+              - type: physical
+                name: eth0
+                subnets:
+                  - type: dhcp
+                  - type: dhcp6
+            EOT"
+
+            sudo sh -c "cat <<EOT >> /etc/rc.local
```
Comment (roydahan) [inline sdcm/cluster_aws.py]: @juliayakovlev why are you doing this?

File: sdcm/cluster_aws.py [hunk 2]
```diff
@@ -306,9 +302,51 @@ def network_config_ipv6_workaround_script():  # pylint: disable=invalid-name
...
+            sh -c 'echo IPV6_AUTOCONF=yes >> /etc/sysconfig/network-scripts/ifcfg-eth0'
```
Comment (roydahan) [inline sdcm/cluster_aws.py]: Also, this is not what I wrote you to add...

I asked for this: 
/etc/sysconfig/network-scripts/init.ipv6-global
sudo ip r a $IPv6_CIDR dev eth0

File: sdcm/cluster_aws.py [hunk 3]
```diff
@@ -578,6 +570,16 @@ def _wait_public_ip(self):
             raise PublicIpNotReady(self._instance)
         LOGGER.debug("[{0._instance}] Got public ip: {0._instance.public_ip_address}".format(self))
 
+    def config_ipv6_as_persistence(self):
+        persistence_ipv6_script = dedent("""
+                        BASE_EC2_NETWORK_URL=http://169.254.169.254/latest/meta-data/network/interfaces/macs/
+                        MAC=`curl -s ${BASE_EC2_NETWORK_URL}`
+                        IPv6_CIDR=`curl -s ${BASE_EC2_NETWORK_URL}${MAC}/subnet-ipv6-cidr-blocks`
+                        sudo sh -c  "echo 'sudo ip r a $IPv6_CIDR dev eth0' >>  /etc/sysconfig/network-scripts/init.ipv6-global"
+                        sudo systemctl restart network
```
Comment (roydahan) [inline sdcm/cluster_aws.py]: Not needed to restart network.
The first time you set the command in the file, it should boot with routing configured.

File: sdcm/cluster_aws.py [hunk 4]
```diff
@@ -578,6 +570,16 @@ def _wait_public_ip(self):
             raise PublicIpNotReady(self._instance)
         LOGGER.debug("[{0._instance}] Got public ip: {0._instance.public_ip_address}".format(self))
 
+    def config_ipv6_as_persistence(self):
+        persistence_ipv6_script = dedent("""
+                        BASE_EC2_NETWORK_URL=http://169.254.169.254/latest/meta-data/network/interfaces/macs/
+                        MAC=`curl -s ${BASE_EC2_NETWORK_URL}`
+                        IPv6_CIDR=`curl -s ${BASE_EC2_NETWORK_URL}${MAC}/subnet-ipv6-cidr-blocks`
+                        sudo sh -c  "echo 'sudo ip r a $IPv6_CIDR dev eth0' >>  /etc/sysconfig/network-scripts/init.ipv6-global"
```
Comment (roydahan) [inline sdcm/cluster_aws.py]: change r a to "route add" so it'll be searchable

Comment (roydahan) [reply sdcm/cluster_aws.py]: Did you verify that it works manually?
(appending to the file this command as follow)

File: sdcm/cluster_aws.py [hunk 5]
```diff
@@ -297,15 +297,6 @@ def network_config_ipv6_workaround_script():  # pylint: disable=invalid-name
...
-            for param in IPV6_AUTOCONF IPV6_DEFROUTE
-            do
-                res=`sudo grep "$param" /etc/sysconfig/network-scripts/ifcfg-eth0`
-                if [[ "${res}x" == "x" ]]
-                then
-                    sudo sh -c  "echo \"$param=yes\" >> /etc/sysconfig/network-scripts/ifcfg-eth0"
-                fi
-            done
-
             sudo ip r a $IPv6_CIDR dev eth0
```
Comment (roydahan) [inline sdcm/cluster_aws.py]: change r a to "route add" so it'll be searchable

**Comment by roydahan** [comment]
@fruch according to @juliayakovlev this is working now.
Can you please review as well?

File: sdcm/cluster_aws.py [hunk 6]
```diff
@@ -578,6 +569,28 @@ def _wait_public_ip(self):
...
         LOGGER.debug("[{0._instance}] Got public ip: {0._instance.public_ip_address}".format(self))
 
+    def config_ipv6_as_persistence(self):
+        cidr = dedent("""
+                        BASE_EC2_NETWORK_URL=http://169.254.169.254/latest/meta-data/network/interfaces/macs/
+                        MAC=`curl -s ${BASE_EC2_NETWORK_URL}`
+                        curl -s ${BASE_EC2_NETWORK_URL}${MAC}/subnet-ipv6-cidr-blocks
+                    """)
+        output = self.remoter.run(f"sudo bash -cxe '{cidr}'")
+        if output.exit_status == 0:
```
Comment (bentsi) [inline sdcm/cluster_aws.py]: no need to check, if exit status will be != 0, remoter throws an exception

File: sdcm/cluster_aws.py [hunk 7]
```diff
@@ -578,6 +569,28 @@ def _wait_public_ip(self):
             raise PublicIpNotReady(self._instance)
         LOGGER.debug("[{0._instance}] Got public ip: {0._instance.public_ip_address}".format(self))
 
+    def config_ipv6_as_persistence(self):
```
Comment (bentsi) [inline sdcm/cluster_aws.py]: persistent - adjective
persistence - noun
`config_ipv6_as_persistence` -> `config_ipv6_as_persistent`

**Comment by juliayakovlev** [comment]
> questions:
> 
> * if the persistence of the IPv6 configuration failed are we able to continue a test?

After reboot the connection to the instance will fail

> * why can't we configure it in a startup script?

Moved to the ```AWSCluster.add_nodes```.
Running test

**Comment by juliayakovlev** [comment]
@bentsi  @roydahan @fruch  Please, review

**Comment by fruch** [comment]
@bentsi it's o.k. it's not in the startup script, this temporary workaround, until the AMI would support ipv6 out of the box.

**Pull request merged**